Sources/Apollo/GraphQLUpload.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+public protocol GraphQLUpload: JSONEncodable {
+  var data: Data { get }
```

Comment:
This property should preferably be a stream instead of actual bytes. For large upload, you don't want to load the entire file into memory at once.

You mentioned you have had a look at https://github.com/Alamofire/Alamofire/blob/master/Source/MultipartFormData.swift ?